docs-v2/customize/guides/advanced/validate-input-and-output.mdx

```diff
@@ -0,0 +1,157 @@
+---
+title: 'Validate input and output'
+sidebarTitle: 'Validate input and output'
+description: 'How to automatically validate your input and ouput with JSON schema and custom alternatives'
+---
+
+Nango provides a straightforward way to validate your data using automatically generated schema files. This guide will walk you through the essentials of data validation in Nango.
+
+## Auto-Generated schema files
+
+When you use Nango CLI, it automatically generates two schema files in the `.nango` folder:
+
+- **schema.ts** a TypeScript file that contains all your models.
+- **schema.json** a JSON Schema file that is used for automatic data validation.
+
+These files can be versioned and integrated into your own codebase, ensuring consistency and reliability across different environments.
+
+## Automatic validation
+
+The `schema.json` file plays a crucial role in Nango's data validation process. This `schema.json` file is used to validate both Actions input/output and Syncs record automatically.
```

Comment:
records plural?

Reply:
Again you're starting with explaining the internals. We want to describe the UX first, with the outcome. Then the internals.